packages/backend/src/apps/lettersg/actions/create-letter/get-data-out-metadata.ts

```diff
@@ -0,0 +1,30 @@
+import { IDataOutMetadata, IExecutionStep } from '@plumber/types'
+
+import { parseS3Id } from '@/helpers/s3'
+
+async function getDataOutMetadata(
+  step: IExecutionStep,
+): Promise<IDataOutMetadata> {
+  const data = step.dataOut
```

Comment:
nit: can zod this too. E.g

https://github.com/opengovsg/plumber/blob/cd8ccd11d77841cdb215581b03dd2dd4870be68c/packages/backend/src/apps/m365-excel/actions/get-table-row/get-data-out-metadata.ts#L13